src/MainWindow.vala

```diff
@@ -469,5 +477,10 @@ namespace PantheonCalculator {
             settings.set_boolean ("extended-shown", button_extended.active);
             settings.set_string ("entry-content", entry.text);
         }
+
+        public void decimal () {
+            var new_decimal = Posix.nl_langinfo (Posix.NLItem.RADIXCHAR); 
```

Comment:
Replace `var` with `unowned string` to avoid string copies.
```suggestion
            unowned string new_decimal = Posix.nl_langinfo (Posix.NLItem.RADIXCHAR); 
```